### DISCUSSION TITLE
[Feature Request] Mobile oxygensource/weldingfuel/powersource overhaul with some optimizations

### DISCUSSION THREAD

**Discussion by BiffW** [FakeFishGames/Barotrauma#10499] [Ideas and suggestions]
Basically unified the way equipments consume oxygen source and welding fuel.
Instead of separately defining consuming speed by identifiers, they consume items with tag "oxygensource"/"weldingtoolfuel" at the same rate.
Instead of directly refering identifiers "oxygentank","weldingfueltank","oxygenitetank","incendiumtank", 
they now use specific tags "oxygensource","weldingtoolfuel","strongoxygensource","strongweldingtoolfuel","strongpowersource" to identify required items, 
which makes it less painful when you want to adjust the durability of an item.

Optimized mobile equipments and devices to encourage usage of advanced items(Oxygenite Tank, Incendium Fuel Tank and Fulgurium Battery)
Now tools with advanced oxygensource/weldingtoolfuel/powersource works with extra effect.
The idea behind this is based on the fact that oxygenite tank gives you speed buff when used with diving suits and fulgurium improves stunning power of a stun baton.
So why shouldn't other equipments have some extra effect with them?

for example:
before:
```
      <StatusEffect type="OnWearing" target="Contained,Character" OxygenAvailable="1000.0" Condition="-0.5" comparison="Or">
        <Conditional IsDead="false" />
        <RequiredItem items="oxygentank" type="Contained" />
      </StatusEffect>
      <StatusEffect type="OnWearing" target="Contained,Character" OxygenAvailable="1500.0" Condition="-0.1" comparison="Or">
        <Conditional IsDead="false" />
        <RequiredItem items="oxygenitetank" type="Contained" />
      </StatusEffect>
      <StatusEffect type="OnWearing" target="Contained,Character" OxygenAvailable="-100.0" Oxygen="-5.0" Condition="-5.0">
        <Conditional IsDead="false" />
        <RequiredItem items="weldingfueltank" type="Contained" />
      </StatusEffect>
      <StatusEffect type="OnWearing" target="Contained,Character" targetlimbs="Torso" OxygenAvailable="-100.0" Oxygen="-7.0" Condition="-5.0">
        <RequiredItem items="incendiumfueltank" type="Contained" />
        <Affliction identifier="burn" amount="3.0" />
      </StatusEffect>
```
after:
```
      <StatusEffect type="OnWearing" target="Contained,Character" OxygenAvailable="1000.0" Condition="-0.3" comparison="Or">
        <Conditional IsDead="false" />
        <RequiredItem items="oxygensource" type="Contained" />
      </StatusEffect>
      <StatusEffect type="OnWearing" target="Contained,Character" OxygenAvailable="500.0" comparison="Or">
        <Conditional IsDead="false" />
        <RequiredItem items="strongoxygensource" type="Contained" />
      </StatusEffect>
      <StatusEffect type="OnWearing" target="Contained,Character" Oxygen="-10.0" Condition="-0.5" interval="1" disabledeltatime="true">
        <Conditional IsDead="false" />
        <RequiredItem items="weldingtoolfuel" type="Contained" />
      </StatusEffect>
      <StatusEffect type="OnWearing" target="Contained,Character" targetlimbs="Torso" interval="1" disabledeltatime="true">
        <Conditional IsDead="false" />
        <RequiredItem items="strongweldingtoolfuel" type="Contained" />
        <Affliction identifier="burn" amount="3.0" />
      </StatusEffect>
```
change list:

Oxygenite Tank:
health changed to 300
added tag "strongoxygensource"
added missing recycle recipe

Incendium Fuel Tank:
health changed to 200
added tag "strongweldingtoolfuel"

Fulgurium Battery Cell:
added tag "strongpowersource"

Welding Tool:
Optimized fuel comsumption
Gain 50% more effectiveness when using a "strongweldingtoolfuel"

Plasma Tool:
Optimized oxygen comsumption
Gain 50% more effectiveness when using a "strongoxygensource"

Flamer, Prototype Steam Cannon:
Added missing talent recipe for flamer
Inflict 100% more burning damage when using a "strongweldingtoolfuel"

Cargo Scooter, Underwater Scooter:
Added gui to switch light state
Light comsumes power separately
Gain 50% more propulsion force when using a "strongpowersource"

Handheld Sonar Beacon:
Now float in water
Can be thrown

Diving Mask, Diving Suit, Abyss Diving Suit, Combat Diving Suit, PUCS, Splitsuit:
Optimized oxygen comsumption

PUCS:
Slowly refill "refillableoxygensource" item when in air.
Only poison user with "weldingtoolfuel" when in water or hull oxygen is low

Exosuit, Funbringer 3000:
Gain 30% more speed when using a "strongoxygensource" just like other diving suits
Added stun resistance(50% probability and 50% reduction)

Exosuit:
Suffer 50% less speed penalty when using a "strongpowersource"

Funbringer 3000:
Suffer 50% less speed penalty when using a "fulguriumfuelrod" or a "fulguriumfuelrodvolatile"

You can download a mod version of these changes below.

**Comment by ubertpendragon** [comment]
Yes, this makes them mod-compatible with custom fuel sources that would share say the speed property of oxygenite

**Comment by BiffW** [comment]
update for 0.20.8.0 (faction branch 100.6.0.0)
[Better Vanilla Equipments.zip](https://github.com/Regalis11/Barotrauma/files/10120942/Better.Vanilla.Equipments.zip)

**Comment by BiffW** [comment]
@Regalis11 At present, Oxygenite Tank, Incendium Fuel, Fulgurium Battery are just bigger capacity version of Oxygen Tank, Welding Fuel Tank and Battery Cell. These features I added can attach some uniqueness to them and provide more reasons for players to use them.
